src/QsCompiler/Compiler/Templates/QirDriverCpp.tt

```diff
@@ -1,16 +1,278 @@
+<# if (entryPointOperation.ContainsArrayType(DataType.RangeType)) { #>
+template<typename T>
+void FreePointerVector(vector<T*>& v)
+{
+    for (auto p : v)
+    {
+        delete p;
+    }
+}
+<# } #>
+
+// This is the function corresponding to the QIR entry-point.
+extern "C" void <#= entryPointOperation.Name #>( // NOLINT
+<# for (int i = 0; i < entryPointOperation.InteropArguments.Count; i++) {
+    var arg = entryPointOperation.InteropArguments[i];
+    Write($"    {arg.CppType()} {arg.Name}");
+    if (i < entryPointOperation.InteropArguments.Count-1) {
+        WriteLine(",");
+    }
+} #>
+
+);
+
+<# if (entryPointOperation.ContainsArgumentType(DataType.BoolType) || entryPointOperation.ContainsArrayType(DataType.BoolType)) { #>
+const char InteropFalseAsChar = 0x0;
+const char InteropTrueAsChar = 0x1;
+map<string, bool> BoolAsCharMap{
+    {"0", InteropFalseAsChar},
+    {"false", InteropFalseAsChar},
+    {"1", InteropTrueAsChar},
+    {"true", InteropTrueAsChar}};
+
+<# } #>
+<# if (entryPointOperation.ContainsArgumentType(DataType.PauliType) || entryPointOperation.ContainsArrayType(DataType.PauliType)) { #>
+map<string, PauliId> PauliMap{
+    {"PauliI", PauliId::PauliId_I},
+    {"PauliX", PauliId::PauliId_X},
+    {"PauliY", PauliId::PauliId_Y},
+    {"PauliZ", PauliId::PauliId_Z}};
+
+<# } #>
+<# if (entryPointOperation.ContainsArgumentType(DataType.ResultType) || entryPointOperation.ContainsArrayType(DataType.ResultType)) { #>
+const char InteropResultZeroAsChar = 0x0;
+const char InteropResultOneAsChar = 0x1;
+map<string, char> ResultAsCharMap{
+    {"0", InteropResultZeroAsChar},
+    {"Zero", InteropResultZeroAsChar},
+    {"1", InteropResultOneAsChar},
+    {"One", InteropResultOneAsChar}
+};
+
+<# } #>
+<# if (entryPointOperation.ContainsArgumentType(DataType.PauliType) || entryPointOperation.ContainsArrayType(DataType.PauliType)) { #>
+char TranslatePauliToChar(PauliId& pauli)
+{
+    return static_cast<char>(pauli);
+}
+
+<# } #>
+<# if (entryPointOperation.ContainsArgumentType(DataType.StringType) || entryPointOperation.ContainsArrayType(DataType.StringType)) { #>
+const char* TranslateStringToCharBuffer(string& s)
+{
+    return s.c_str();
+}
+
+<# } #>
 int main(int argc, char* argv[])
 {
-    return 0;
+    CLI::App app("QIR Standalone Entry Point Inputs Reference");
+
+    // Initialize simulator.
+    unique_ptr<IRuntimeDriver> sim = CreateFullstateSimulator();
+    QirContextScope qirctx(sim.get(), false /*trackAllocatedObjects*/);
+
+    // Add the --simulation-output options.
+    // N.B. This option should be present in all standalone drivers.
+    string simulationOutputFile;
+    CLI::Option* simulationOutputFileOpt = app.add_option(
+        "--simulation-output", simulationOutputFile,
+        "File where the output produced during the simulation is written");
+    
+<# foreach (var arg in entryPointOperation.InteropArguments) {
+    WriteLine("");
+    WriteLine($"    {arg.CppVarType()} {arg.Name};"); 
```

Comment:
For indentation, it would be a bit more readable to have a global `Indentation` string that you could reference.

Reply:
not sure what is meant for this, in some cases multiple indents are necessary, so could look a little odd that way

Reply:
Yes, this might be a matter of personal preference so it's ok to leave it as it is.